src/dfa.rs

```diff
@@ -225,24 +225,23 @@ impl<T> Result<T> {
     }
 }
 
+
 /// State is a DFA state. It contains an ordered set of NFA states (not
 /// necessarily complete) and a smattering of flags.
 ///
+/// The flags are packed into the first byte of data.
+///
 /// States don't carry their transitions. Instead, transitions are stored in
 /// a single row-major table.
+///
+/// Delta encoding is used to store the instruction pointers.
+/// The first instruction pointer is stored directly starting
+/// at data[1], and each following pointer is stored as an offset
+/// to the previous one. If a delta is in the range -127..127,
+/// it is packed into a single byte; Otherwise the byte 128 (-128 as an i8)
+/// is coded as a flag, followed by 4 bytes encoding the delta.
 #[derive(Clone, Eq, Hash, PartialEq)]
-struct State {
-    /// The set of NFA states in this DFA state, which are computed by
-    /// following epsilon transitions from `insts[0]`. Note that not all
-    /// epsilon transitions are necessarily followed! Namely, epsilon
-    /// transitions that correspond to empty assertions are only followed if
-    /// the flags set at the current byte satisfy the assertion.
-    insts: Box<[InstPtr]>,
-    /// Flags for this state, which indicate whether it is a match state,
-    /// observed an ASCII word byte and whether any instruction in `insts`
-    /// is a zero-width assertion.
-    flags: StateFlags,
-}
+struct State{data: Box<[u8]>}
```

Comment:
Could you please use the same style of formatting as the rest of the code? Probably `struct State(Box<[u8]>)` would be nice here.